collated/main/A0141128R.md

```diff
@@ -0,0 +1,692 @@
+                new DueByTime(newDueByTime),
+                new Priority(newPriority),
+                new UniqueTagList(tagSet)
+        );
+        this.floating = floating;
+        this.targetIndex = targetIndex;
+        this.removePriority = removePriority;
+    }
+        
+    /**
+     * For RANGE DATE AND TIME
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     * 
+```
+###### /java/seedu/cmdo/logic/commands/EditCommand.java
+``` java
+     */
+    public EditCommand(boolean removePriority, 
+    				  int targetIndex,
+    		          String details,
+                      LocalDate dueByDateStart,
+                      LocalTime dueByTimeStart,
+                      LocalDate dueByDateEnd,
+                      LocalTime dueByTimeEnd,
+                      String priority,
+                      Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEditWith = new Task(
+                new Detail(details),
+                new DueByDate (dueByDateStart, dueByDateEnd),
+                new DueByTime(dueByTimeStart, dueByTimeEnd),
+                new Priority(priority),
+                new UniqueTagList(tagSet)
+        );
+        this.targetIndex = targetIndex;
+        this.isUndoable = true;
+        floating = false;//since if range constructor is used, user would have keyed in a timing 
+        this.removePriority = removePriority; 
+    }
+    
+    public ReadOnlyTask getTask() {
+        return toEditWith;
+    }
+
+```
+###### /java/seedu/cmdo/logic/commands/EditCommand.java
+``` java
+    @Override
+    public CommandResult execute() {
+        UnmodifiableObservableList<ReadOnlyTask> lastShownList = model.getFilteredTaskList();
+
+        // Check if target index is valid
+        if (lastShownList.size() < targetIndex) {
+            indicateAttemptToExecuteIncorrectCommand();
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+        
+        // Retrieve the task and check if done.
+        ReadOnlyTask taskToEdit = lastShownList.get(targetIndex - 1);
+        if (taskToEdit.checkDone().value) {
+            indicateAttemptToExecuteIncorrectCommand();
+        	return new CommandResult(Messages.MESSAGE_EDIT_TASK_IS_DONE_ERROR);
+        }
+        
+        //write a method for it
+        //check for changes in detail and append
+        if(toEditWith.getDetail().toString().equals(""))
+        	toEditWith.setDetail(taskToEdit.getDetail());
+        
+        //check if changing to floating task
+        if(floating)
+        	toEditWith.setFloating();
+        //check for if time is empty and append and check if have changes in date otherwise append old date
+        else{
+        if(toEditWith.getDueByDate().dateNotEntered() && toEditWith.getDueByTime().timeNotEntered()){
```

Comment:
coding standard violation;, Avoid lines longer than 80 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.